cpp_test_suite/new_tests/cxx_group.cpp

```diff
@@ -827,6 +827,39 @@ class GroupTestSuite: public CxxTest::TestSuite
 		CxxTest::TangoPrinter::restore_unset("double_attr_value");
 	}
 
+	/* Verifies that a group can contain devices from a remote TANGO_HOST
+	 * (a Tango instance different from client's default TANGO_HOST).
+	 * An issue was reported when resolving names containing wildcards. */
+
+	void test_use_devices_from_remote_tango_host()
+	{
+		const std::string original_tango_host = std::getenv("TANGO_HOST");
+		const std::string external_tango_host = std::getenv("TANGO_HOST2");
+
+		const bool force_update = true;
+
+		TS_ASSERT_EQUALS(0, setenv("TANGO_HOST", external_tango_host.c_str(), force_update));
+		ApiUtil::instance()->cleanup();
+
+		Group group("group");
+		group.add("tango://" + original_tango_host + "/" + device1_name + "*");
```

Comment:
Nice to have set up a test with 2 tango databases.
At first sight, it looks like a device named device1_name exists in both Tango control systems (tests are executed with "--device1=test/debian8/10" parameter).
It would have been interesting to have a way to ensure that we are really sending the state command to the right device.
Talking to at least 1 device having its name in only one DB would have been interesting.
Another way could be to ensure that the devices $TANGO_HOST/test/debian8/10 and $TANGO_HOST2/test/debian8/10 return different values for the state.... or to read an attribute for which we could set a specific value before the test

Reply:
No, the device is defined only in the first (original) system (in `tango_cs` container).
The second one (`tango_cs2` container) is empty.

The devices are created by `conf_devtest` program invoked from `pre_test.sh` script just once, with TANGO_HOST pointing to the original system.

Please check the travis log for the commit that added the test:

```
Exception : 

Tango exception
Severity = ERROR 
Error reason = DB_DeviceNotDefined
Desc : device test/debian8/10 not defined in the database !
Origin : DataBase::ImportDevice()

...

In GroupTestSuite::test_use_devices_from_remote_tango_host:
../../new_tests/cxx_group.cpp:834: Error: Test failed: Exception thrown from test

TEST FAILED in:
1 iteration of test case
1 iteration of test suite
```

You can also verify this using the example provided in #495 after running `make start-tango`. It is failing without the correction:
```
TANGO_HOST=$TANGO_HOST2 LD_LIBRARY_PATH=~/Documents/cppTango/build ~/Documents/grouptest/a.out "tango://$TANGO_HOST/test/debian*/*"
```

Reply:
> No, the device is defined only in the first (original) system (in `tango_cs` container).
> The second one (`tango_cs2` container) is empty. 
> The devices are created by `conf_devtest` program invoked from `pre_test.sh` script just once, with TANGO_HOST pointing to the original system.

Good. I missed that part indeed.

> You can also verify this using the example provided in #495 after running `make start-tango`. It is failing without the correction

This I already did and indeed #495 seems to be solved with this PR.